### PR TITLE
Fix of tf.GenericFunction conversion in convert_model()

### PR DESCRIPTION
**Pull request by popovaan**
Root cause analysis: 
tf.GenericFunction test was initially broken which led to improper testing of the conversion of tf.GenericFunction.

Solution:
Fixed tf.GenericFunction test, fixed conversion of tf.GenericFunction.
Added support of conversion of tf.Graph.

Ticket: 106247

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests - Tests for this case will be done in separate PR to MO args tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py [hunk 1]
```diff
@@ -269,21 +269,14 @@ def create_tf_checkpoint(tmp_dir):
 def create_tf_function(temp_dir):
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:269]: I think it makes sense to have functions and tests for both `keras.Model` and `ConcreteFunction`

File: tools/mo/openvino/tools/mo/front/tf/loader.py [hunk 2]
```diff
@@ -226,6 +226,14 @@ def tf_function(x):
 
         conc_func = tf_function.get_concrete_function(model_inputs)
         return freeze_tf2_concrete_function(model, conc_func, env_setup)
+    if isinstance(model, tf.types.experimental.GenericFunction):
+        env_setup = get_environment_setup("tf")
+
+        assert hasattr(model, "input_signature") and model.input_signature is not None, \
+            "'input_signature' needs to be set for model conversion."
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/loader.py:233]: can we have a test for this case with `input_signature` when obtaining of `ConcreteFunction` from `GenericFunction` is not possible

File: tools/mo/openvino/tools/mo/front/tf/loader.py [hunk 3]
```diff
@@ -226,6 +226,14 @@ def tf_function(x):
 
         conc_func = tf_function.get_concrete_function(model_inputs)
         return freeze_tf2_concrete_function(model, conc_func, env_setup)
+    if isinstance(model, tf.types.experimental.GenericFunction):
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/loader.py:229]: It looks that `tf.types.experimental.GenericFunction` is introduced recently in `2.4`. Can we leave MO compatible with early TF versions please?

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py (hunk 1 above)
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:269]: We have keras.Model test here: https://github.com/openvinotoolkit/openvino/blob/656d7fe380e8a5d8c245a47c2c0e41b56baddeb7/tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py#L40

File: tools/mo/openvino/tools/mo/front/tf/loader.py (hunk 2 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/front/tf/loader.py:232]: Done.

File: tools/mo/openvino/tools/mo/front/tf/loader.py (hunk 3 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/front/tf/loader.py]: I double checked it, `GenericFunction `and `ConcreteFunction `was introduced in 2.6. I added checks of TF version for these types.

Comment (rkazants) [reply tools/mo/openvino/tools/mo/front/tf/loader.py]: this is experimental type. May be it makes sense to check stable API. Is it possible? Please check

**Review by rkazants — APPROVED** [review]

**Pull request merged**